src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1479,6 +1482,120 @@ class Simplex_tree {
+   * this->initialize_filtration, and this->compute_persistent_cohomology.
+   * @return A vector of four persistence diagrams. The first one is Ordinary, the 
+   * second one is Relative, the third one is Extended+ and the fourth one is Extended-.
+   * See section 2.2 in https://link.springer.com/article/10.1007/s10208-017-9370-z for a description of these subtypes.
+   */
+  std::vector<std::vector<std::pair<int, std::pair<double, double>>>> compute_extended_persistence_subdiagrams(const std::vector<std::pair<int, std::pair<double, double>>>& dgm){
+    std::vector<std::vector<std::pair<int, std::pair<double, double>>>> new_dgm(4); 
+    double x, y;
+    double minval_ = this->minval_;
+    double maxval_ = this->maxval_;
+    for(unsigned int i = 0; i < dgm.size(); i++){ 
+      int h = dgm[i].first; 
+      double px = dgm[i].second.first; 
+      double py = dgm[i].second.second;
+      if(std::isinf(py))  continue;
+      else{
+        if ((px <= -1) & (py <= -1)){
+          x = minval_ + (maxval_-minval_)*(px + 2); 
+          y = minval_ + (maxval_-minval_)*(py + 2); 
+          new_dgm[0].push_back(std::make_pair(h, std::make_pair(x,y))); 
+        }
+        if ((px >=  1) & (py >=  1)){
+          x = minval_ - (maxval_-minval_)*(px - 2); 
+          y = minval_ - (maxval_-minval_)*(py - 2); 
+          new_dgm[1].push_back(std::make_pair(h, std::make_pair(x,y))); 
+        }
+        if ((px <= -1) & (py >=  1)){
+          x = minval_ + (maxval_-minval_)*(px + 2); 
+          y = minval_ - (maxval_-minval_)*(py - 2);
+          if (x <= y){
+            new_dgm[2].push_back(std::make_pair(h, std::make_pair(x,y)));
+          }
+          else{  
+            new_dgm[3].push_back(std::make_pair(h, std::make_pair(x,y)));
+          }
+        }
+      }
+    }
+    return new_dgm;
+  }
+
+  /** \brief Extend filtration for computing extended persistence. 
+   * This function only uses the filtration values at the 0-dimensional simplices, 
+   * and computes the extended persistence diagram induced by the lower-star filtration 
+   * computed with these values. 
+   * \post Note that after calling this function, the filtration 
+   * values are actually modified. The function compute_extended_persistence_subdiagrams 
+   * retrieves the original values and separates the extended persistence diagram points 
+   * w.r.t. their types (Ord, Rel, Ext+, Ext-) and should always be called after 
+   * computing the persistent homology of the extended simplicial complex.
+   * \post Note that this code creates an extra vertex internally, so you should make sure that
+   * the Simplex tree does not contain a vertex with the largest Vertex_handle. 
+   */
+  void extend_filtration() {
+
+    // Compute maximum and minimum of filtration values
+    int maxvert = std::numeric_limits<int>::min();
+    this->minval_ = std::numeric_limits<double>::max();
+    this->maxval_ = std::numeric_limits<double>::min();
```

Comment:
`lowest`, not `min`. `min` gives you the smallest **positive** double.
Shouldn't all the `double` be `Filtration_value`?

Reply:
Thanks. Done.